src/fabric/mod.rs

```diff
@@ -245,35 +254,52 @@ impl Fabric {
 #[derive(Clone, Debug, Copy, PartialEq, Default, Hash, Eq, Ord, PartialOrd)]
 pub struct UniqueId(usize);
 
-#[derive(Clone, Debug, Copy)]
-pub struct Link {
-    ideal: f32,
-    material: usize,
-}
-
-const DEFAULT_MATERIALS: [Material; 2] = [
+const MATERIALS: [Material;5] = [
     Material {
+        name: ":push",
         role: Push,
         stiffness: 3.0,
         mass: 1.0,
     },
     Material {
+        name: ":pull",
         role: Pull,
         stiffness: 1.0,
         mass: 0.1,
     },
+    Material {
+        name: ":bow-tie",
+        role: Pull,
+        stiffness: 0.7,
+        mass: 0.1,
+    },
+    Material {
+        name: ":north",
+        role: Pull,
+        stiffness: 0.5,
+        mass: 0.01,
+    },
+    Material {
+        name: ":south",
+        role: Pull,
+        stiffness: 0.5,
+        mass: 0.01,
+    }
 ];
 
-const DEFAULT_PUSH_MATERIAL: usize = 0;
-const DEFAULT_PULL_MATERIAL: usize = 1;
+#[derive(Clone, Debug)]
+pub struct Link {
+    pub ideal: f32,
+    pub material: String,
```

Comment:
```suggestion
    pub material_name: String,
```
And the resulting usage renames as well.

This disambiguates it from the Material struct